src/AbstractConfig.php

```diff
@@ -129,7 +129,25 @@ public function set($key, $value)
      */
     public function has($key)
     {
-        return !is_null($this->get($key));
+        // Check if already cached
```

Comment:
Since this is effectively copying the current body of `get()`, I'd suggest letting `has()` prime the cache when the key is found.  You can then remove all this similar logic from `get()`, and just have `get()` call out to `has()` to see if it exists.  If `has()` says false, then `get()` returns `$default`... if it says TRUE, then `get()` can return it right out of the cache.